### ISSUE TITLE
allow configuration of securityContext at pod-level

### ISSUE THREAD

**Issue by scott-grimes** [rstudio/helm#195]
Our cluster has strict limitations around what user ids are permitted and what capabilities are allowed (no root users allowed, etc). 

For rstudio-connect chart version 0.2.27 with `securityContext.privileged: false` we get the following error:

```
Warning Failed 11s (x4 over 14s) kubelet Error: container has runAsNonRoot and image has a non-numeric user (nobody), cannot verify user is non-root
```

We can configure the user explicitly: 

```
securityContext:
  privileged: false
  runAsUser: 1337
```

But this results in a permission denied error on boot in the `connect` container:

```
#----> prestart.bash <date>: Ensuring /mnt/dynamic/rstudio exists
mkdir: cannot create directory '/mnt/dynamic': Permission denied
```
The securityContext is configurable for the container at https://github.com/rstudio/helm/blob/main/charts/rstudio-connect/templates/deployment.yaml#L90-L93, but we can't configure the securityContext ad the pod-level (which would allow us to modify mount ownership)

**Comment by scott-grimes** [comment]
fixed by https://github.com/rstudio/helm/pull/196

**Comment by colearendt** [comment]
Thanks for the feedback and PR! Do you mind sharing whether you are using the launcher beta here? And what type of Kubernetes cluster /distro? (I am 99% sure you've told me this before, but I'm blanking right now).

This is a good point though, related to #176 (but in a Connect context), I think the `/mnt/dynamic` piece is actually a piece of this startup script that is left over from Workbench. There is some cleanup that we can do here for sure.

https://github.com/rstudio/helm/blob/ac2ad2b2e3c84309a456f6229934e5170c82164e/charts/rstudio-connect/prestart.bash#L7

That said, I worry that you are just going to keep running into problems running as "non root" in the container. Connect (when using the launcher) knows how to relax its sandboxing in the kernel (bind mounts and other items that require `CAP_SYS_ADMIN`). That said, the service still presumes that it will be able to do things that only `root` can do (run processes as other users, etc.). We have a backlog item to relax a lot of these items, but there's a lot to trace through before we expect that it will work. 🙈

**Comment by colearendt** [comment]
Merged #196
